exchanges/okcoin/okcoin_websocket.go

```diff
@@ -108,27 +169,66 @@ func (o *OKCoin) WsReadData() {
 }
 
 // WsHandleData will read websocket raw data and pass to appropriate handler
-func (o *OKCoin) WsHandleData(respRaw []byte) error {
+func (o *Okcoin) WsHandleData(respRaw []byte) error {
+	if bytes.Equal(respRaw, []byte(pongBytes)) {
+		return nil
+	}
 	var dataResponse WebsocketDataResponse
 	err := json.Unmarshal(respRaw, &dataResponse)
 	if err != nil {
 		return err
 	}
+	if dataResponse.ID != "" {
+		if !o.Websocket.Match.IncomingWithData(dataResponse.ID, respRaw) {
+			return fmt.Errorf("couldn't match incoming message with id: %s and operation: %s", dataResponse.ID, dataResponse.Operation)
+		}
+		return nil
+	}
 	if len(dataResponse.Data) > 0 {
-		switch o.GetWsChannelWithoutOrderType(dataResponse.Table) {
-		case okcoinWsCandle60s, okcoinWsCandle180s, okcoinWsCandle300s,
-			okcoinWsCandle900s, okcoinWsCandle1800s, okcoinWsCandle3600s,
-			okcoinWsCandle7200s, okcoinWsCandle14400s, okcoinWsCandle21600s,
-			okcoinWsCandle43200s, okcoinWsCandle86400s, okcoinWsCandle604900s:
-			return o.wsProcessCandles(respRaw)
-		case okcoinWsDepth, okcoinWsDepth5:
-			return o.WsProcessOrderBook(respRaw)
-		case okcoinWsTicker:
+		switch dataResponse.Arguments.Channel {
+		case wsInstruments:
+			return o.wsProcessInstruments(respRaw)
+		case wsTickers:
 			return o.wsProcessTickers(respRaw)
-		case okcoinWsTrade:
+		case wsCandle3M, wsCandle1M, wsCandle1W, wsCandle1D, wsCandle2D, wsCandle3D, wsCandle5D,
+			wsCandle12H, wsCandle6H, wsCandle4H, wsCandle2H, wsCandle1H, wsCandle30m, wsCandle15m,
+			wsCandle5m, wsCandle3m, wsCandle1m, wsCandle3Mutc, wsCandle1Mutc, wsCandle1Wutc, wsCandle1Dutc,
+			wsCandle2Dutc, wsCandle3Dutc, wsCandle5Dutc, wsCandle12Hutc, wsCandle6Hutc:
+			return o.wsProcessCandles(respRaw)
+		case wsTrades:
 			return o.wsProcessTrades(respRaw)
-		case okcoinWsOrder:
-			return o.wsProcessOrder(respRaw)
+		case wsOrderbooks,
+			wsOrderbooksL5,
+			wsOrderbookL1,
+			wsOrderbookTickByTickL400,
+			wsOrderbookTickByTickL50:
+			return o.wsProcessOrderbook(respRaw, dataResponse.Arguments.Channel)
+		case wsStatus:
+			var resp WebsocketStatus
+			err = json.Unmarshal(respRaw, &resp)
+			if err != nil {
+				return err
+			}
+			for x := range resp.Data {
+				systemStatus := fmt.Sprintf("%s %s on system %s %s service type\nFrom %s\nTo %s\n", systemStateString(resp.Data[x].State), resp.Data[x].Title, resp.Data[x].System, systemStatusServiceTypeString(resp.Data[x].ServiceType), resp.Data[x].Begin.Time().String(), resp.Data[x].End.Time().String())
```

Comment:
Can we please remove the new lines as this seems like an oversight:

<img width="977" alt="image" src="https://github.com/thrasher-corp/gocryptotrader/assets/9391715/624faa69-e56d-47b8-ba64-e92437388012">